### ISSUE TITLE
Areaspline is connecting null points when added via chart.addSeries()

### ISSUE THREAD

**Issue by mnesterenkons** [highcharts/highcharts#6292]
#### Expected behaviour
Calling `chart.addSeries(...)` with `connectNulls` as `false` should not draw a line between null points when a series replaces another.

#### Actual behaviour
Calling `chart.addSeries(...)` draws a line between two points which have points inbetween them with `null` values. This also affects `chart.update(...)` if replacing the series.

#### Live demo with steps to reproduce
JSFiddle: http://jsfiddle.net/2gzxg2qq/1/

The first series in this example has a sharp cliff near the end. Goes from a value of 4 to 0.  The series is then replaced by another series without any null points. One second after that, the series is replaced with yet another series with the same data as the initial one. We now see a curve going from the beginning of the chart to the end, which was not there when we saw the first series.

Note: The second series "Test 2" is necessary to reproduce this, though I'm not sure why. If I replace the first series with the third, this issue is not reproducible.

#### Affected browser(s)
Chrome, Firefox, IE

**Comment by pawelfus** [comment]
Thank you for reporting!

I was hoping to use `setData()`, but it looks like the second series is broken in this case: http://jsfiddle.net/2gzxg2qq/2/

**Comment by mnesterenkons** [comment]
I found a workaround short of destroying the chart and re-rendering it. Before we set the third series, we can set an empty series. Essentially `series: [{ data: [] }]`. As long as we redraw the chart for the empty series, we can set the third series and redraw again without an issue. Not ideal, but it works.

http://jsfiddle.net/2gzxg2qq/3/

I replaced the the `series.remove()` and `chart.addSeries` calls with `chart.update()` because I think it looks a bit cleaner. This workaround works for both ways of adding/replacing a series though.

**Comment by TorsteinHonsi** [comment]
@pawelfus It happens because the stack items are not destroyed after removing the first series, instead they are re-used when adding the new one.

You previously added the `stacksTouched` flag for situations when a point was deleted. In this case the whole series is deleted. Here's a proposed fix:

```js
/**
 * Set all the stacks to initial states and destroy unused ones.
 */
Axis.prototype.resetStacks = function () {
	var stacks = this.stacks,
		type,
		i;
	if (!this.isXAxis) {
		for (type in stacks) {
			for (i in stacks[type]) {

				// Clean up memory after point deletion (#1044, #4320)
				if (stacks[type][i].touched < this.stacksTouched) {
					stacks[type][i].destroy();
					delete stacks[type][i];

				// Reset stacks
				} else {
					stacks[type][i].total = null;
					stacks[type][i].cum = null;
				}
			}
		}
		this.stacksTouched++; // <= Fixes #6292
	}
};
```

Do you see any problem with this, or is it too aggressive? Will this result in stacks being deleted too often when they don't have to?

**Comment by TorsteinHonsi** [comment]
Sorry, apperently _I_ added the stacksTouched flag: https://github.com/highcharts/highcharts/commit/18042ed83219110c33ed67b02d612d5e281d5c69. 

> Do I see any problems with it?

Not sure, I have to run more tests.
